Adds pronoun, pride pins; allows more accessories

Sweedle opened the pull request:
:cl: Sweedle & Rootoo807
rscadd: Adds pronoun badges, found in the "accessories" tab of the loadout.
rscadd: Adds pride pins, found in the "accessories" tab of the loadout.
tweak: Ranks and pins can now be attached to labcoats.
/:cl:

Pronoun badges:
![image](https://user-images.githubusercontent.com/12488672/102283223-d0b73580-3ee6-11eb-930d-e37c15c74300.png)

Pride pins:
![image](https://user-images.githubusercontent.com/12488672/102283252-e4629c00-3ee6-11eb-92ce-c6f003b624c9.png)

Torch said trans rights!

<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You'll find a README and example file in .\html\changelogs\ for further instructions.

You can also find a template for adding your changelog directly to the PR description here: https://github.com/Baystation12/Baystation12/wiki/Automatic-changelog-generation
-->

Lorwp commented on code/modules/clothing/under/accessories/pridepins.dm:
```diff
@@ -0,0 +1,42 @@
+obj/item/clothing/accessory/pridepins/
```
```suggestion
/obj/item/clothing/accessory/pridepins/
```

Sweedle replied:
Resolved this and other similar stuff. Thanks!

Lorwp commented on code/modules/clothing/under/accessories/pridepins.dm:
```diff
@@ -0,0 +1,42 @@
+	on_rolled = list("down" = "none")
+	slot = ACCESSORY_SLOT_INSIGNIA
+
+obj/item/clothing/accessory/pridepins/lesbianpin
```
```suggestion
/obj/item/clothing/accessory/pridepins/lesbianpin
```

Lorwp commented on code/modules/clothing/under/accessories/pridepins.dm:
```diff
@@ -0,0 +1,42 @@
+	desc = "A pin celebrating lesbian pride."
+	icon_state = "lesbianpin"
+
+obj/item/clothing/accessory/pridepins/bisexualpin
```
```suggestion
/obj/item/clothing/accessory/pridepins/bisexualpin
```

Lorwp commented on code/modules/clothing/under/accessories/pridepins.dm:
```diff
@@ -0,0 +1,42 @@
+	desc = "A pin celebrating bisexual pride."
+	icon_state = "bisexualpin"
+
+obj/item/clothing/accessory/pridepins/gaypin
```
```suggestion
/obj/item/clothing/accessory/pridepins/gaypin
```

Lorwp commented on code/modules/clothing/under/accessories/pridepins.dm:
```diff
@@ -0,0 +1,42 @@
+	desc = "A pin celebrating gay pride."
+	icon_state = "gaypin"
+
+obj/item/clothing/accessory/pridepins/pansexualpin
```
```suggestion
/obj/item/clothing/accessory/pridepins/pansexualpin
```

Lorwp commented on code/modules/clothing/under/accessories/pridepins.dm:
```diff
@@ -0,0 +1,42 @@
+	desc = "A pin celebrating pansexual pride."
+	icon_state = "pansexualpin"
+
+obj/item/clothing/accessory/pridepins/nonbinarypin
```
```suggestion
/obj/item/clothing/accessory/pridepins/nonbinarypin
```

Lorwp commented on code/modules/clothing/under/accessories/pridepins.dm:
```diff
@@ -0,0 +1,42 @@
+	desc = "A pin celebrating nonbinary pride."
+	icon_state = "nonbinarypin"
+
+obj/item/clothing/accessory/pridepins/asexualpin
```
```suggestion
/obj/item/clothing/accessory/pridepins/asexualpin
```

Lorwp commented on code/modules/clothing/under/accessories/pridepins.dm:
```diff
@@ -0,0 +1,42 @@
+	desc = "A pin celebrating asexual pride."
+	icon_state = "asexualpin"
+
+obj/item/clothing/accessory/pridepins/intersexpin
```
```suggestion
/obj/item/clothing/accessory/pridepins/intersexpin
```

Lorwp commented on code/modules/clothing/under/accessories/pronouns.dm:
```diff
@@ -0,0 +1,39 @@
+obj/item/clothing/accessory/pronouns/
```
```suggestion
/obj/item/clothing/accessory/pronouns/
```

Lorwp commented on code/modules/clothing/under/accessories/pronouns.dm:
```diff
@@ -0,0 +1,39 @@
+	slot = ACCESSORY_SLOT_INSIGNIA
+	high_visibility = 1
+
+obj/item/clothing/accessory/pronouns/hehimbadge
```
```suggestion
/obj/item/clothing/accessory/pronouns/hehimbadge
```

Lorwp commented on code/modules/clothing/under/accessories/pronouns.dm:
```diff
@@ -0,0 +1,39 @@
+	icon_state = "pronounpin"
+
+
+obj/item/clothing/accessory/pronouns/sheherbadge
```
```suggestion
/obj/item/clothing/accessory/pronouns/sheherbadge
```

Lorwp commented on code/modules/clothing/under/accessories/pronouns.dm:
```diff
@@ -0,0 +1,39 @@
+	desc = "A badge showing the wearer's pronouns: she, hers and hers."
+	icon_state = "pronounpin"
+
+obj/item/clothing/accessory/pronouns/hetheybadge
```
```suggestion
/obj/item/clothing/accessory/pronouns/hetheybadge
```

Lorwp commented on code/modules/clothing/under/accessories/pronouns.dm:
```diff
@@ -0,0 +1,39 @@
+	desc = "A badge showing the wearer's pronouns: he, him and his or they, them and theirs."
+	icon_state = "pronounpin"
+
+obj/item/clothing/accessory/pronouns/shetheybadge
```
```suggestion
/obj/item/clothing/accessory/pronouns/shetheybadge
```

Lorwp commented on code/modules/clothing/under/accessories/pronouns.dm:
```diff
@@ -0,0 +1,39 @@
+	desc = "A badge showing the wearer's pronouns: she, her and hers or they, them and theirs."
+	icon_state = "pronounpin"
+
+obj/item/clothing/accessory/pronouns/zehirbadge
```
```suggestion
/obj/item/clothing/accessory/pronouns/zehirbadge
```

Lorwp commented on code/modules/clothing/under/accessories/pronouns.dm:
```diff
@@ -0,0 +1,39 @@
+	desc = "A badge showing the wearer's pronouns: ze, hir and hirs."
+	icon_state = "pronounpin"
+
+obj/item/clothing/accessory/pronouns/pleaseaskbadge
```
```suggestion
/obj/item/clothing/accessory/pronouns/pleaseaskbadge
```

Spookerton commented on code/modules/client/preference_setup/loadout/lists/accessories.dm:
```diff
@@ -127,3 +127,35 @@
+/datum/gear/accessory/pronouns/New()
+	..()
+	var/pronouns = list()
+	pronouns["they/them"] = /obj/item/clothing/accessory/pronouns/
```
```suggestion
	pronouns["they/them"] = /obj/item/clothing/accessory/pronouns
```

Sweedle replied:
All these resolved; stuff in the accessories list is irrelevant per POD's review

PsiOmegaDelta commented on code/modules/client/preference_setup/loadout/lists/accessories.dm:
```diff
@@ -127,3 +127,35 @@
+	display_name = "pride pin selection"
+	path = /obj/item/clothing/accessory/pridepins
+
+/datum/gear/accessory/pridepins/New()
```
Looks like you could delete this entire override and just use `flags = GEAR_HAS_TYPE_SELECTION` on line 149 instead

PsiOmegaDelta commented on code/modules/client/preference_setup/loadout/lists/accessories.dm:
```diff
@@ -127,3 +127,35 @@
+	display_name = "pronoun badge selection"
+	path = /obj/item/clothing/accessory/pronouns
+
+/datum/gear/accessory/pronouns/New()
```
Same as with the pins

Sweedle replied:
To make sure I've got this right, delete 134-143 (all the list stuff), but _not_ `gear_tweaks += new/datum/gear_tweak/path(pridepins)` and replace it with `flags = GEAR_HAS_TYPE_SELECTION`?

Sweedle replied:
Yup, thanks, Sweedle! These have been fixed

Spookerton commented on code/modules/clothing/under/accessories/pronouns.dm:
```diff
@@ -0,0 +1,39 @@
+/obj/item/clothing/accessory/pronouns/
```
```suggestion
/obj/item/clothing/accessory/pronouns
```

Spookerton commented on code/modules/clothing/under/accessories/pronouns.dm:
```diff
@@ -0,0 +1,39 @@
+/obj/item/clothing/accessory/pronouns/pleaseaskbadge
+	name = "ask my pronouns badge"
+	desc = "A badge asking others to ask for the wearer's pronouns."
+	icon_state = "pronounpin"
```
```suggestion
	icon_state = "pronounpin"

```

Spookerton commented on code/modules/clothing/under/accessories/pridepins.dm:
```diff
@@ -0,0 +1,42 @@
+/obj/item/clothing/accessory/pridepins/intersexpin
+	name = "intersex pride pin"
+	desc = "A pin celebrating intersex pride."
+	icon_state = "intersexpin"
```
```suggestion
	icon_state = "intersexpin"

```

Spookerton commented on code/modules/client/preference_setup/loadout/lists/accessories.dm:
```diff
@@ -127,3 +127,35 @@
+	pridepins["nonbinary pride pin"] = /obj/item/clothing/accessory/pridepins/nonbinarypin
+	pridepins["asexual pride pin"] = /obj/item/clothing/accessory/pridepins/asexualpin
+	pridepins["intersex pride pin"] = /obj/item/clothing/accessory/pridepins/intersexpin
+	gear_tweaks += new/datum/gear_tweak/path(pridepins)
```
```suggestion
	gear_tweaks += new/datum/gear_tweak/path(pridepins)

```

Spookerton reviewed (changes requested):
style stuff - no trailing slashes on paths, newlines at file ends, etc.

Sweedle commented:
[Squashed! ](https://www.youtube.com/watch?v=gKQOXYB2cd8)

Sweedle committed:
Adds pronoun, pride pins; allows more accessories

Adds pronoun badges, pride pins.

Also allows more accessories on labcoats; armor.

Includes pride pins, pronouns

Forgot to do this

Fixes / on item paths

Makes requested changes

Now to figure out how to squash...

Adds pronoun, pride pins; allows more accessories

Adds pronoun badges, pride pins.

Also allows more accessories on labcoats; armor.

Includes pride pins, pronouns

Forgot to do this

Fixes / on item paths

Makes requested changes

Now to figure out how to squash...

Delete unnecessary icon states

Squash practice 2.0.

derpymcfuckwit commented:
can u have a one with two lightning symbols that'd be cool i think

trollmarcus commented:
I love this idea! Should serve well as an indicator who to be griefed first.
